numpy/lib/_iotools.py

```diff
@@ -50,6 +54,13 @@ def _is_bytes_like(obj):
     return True
 
 
+def _is_path(obj):
+    """
+    Check whether obj is a pathlib.Path object.
+    """
+    return PurePath is not None and isinstance(obj, PurePath)
```

Comment:
Hmm, that is an interesting thought, but I just plugged it into the interpreter and it gives a `TypeError` for `isinstance(x, None)`:

```python
>>> p=Path('x')
>>> isinstance(p, None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: isinstance() arg 2 must be a type or tuple of types
```

Note that `PurePath` will be `None` if pathlib is not installed, so the `PurePath is None` is protecting that `isinstance`.